tikv/tikv
# raftstore: support dynamically scaling pool size (#11159)

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>



### What problem does this PR solve?

Issue Number: close #11159  



### What is changed and how it works?

Proposal: #11159 

What's Changed: #11159 

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test
- Manual test
   - Run ```tiup playground``` to setup a cluster
   - Run ```mysql -h127.0.0.1 -uroot -P 4000``` to connect to the tidb
   - Run ```show config where type='tikv' and name='raftstore.store-pool-size';``` to query current configuration
   - Run ```set config tikv `raftstore.apply-pool-size`=6;``` to increase the pool size
   - Run ```show config where type='tikv' and name='raftstore.store-pool-size';``` to check current configuration
   - Run ```ps -T -p `pidof tikv-server` | grep raftstore-``` to check poller threads
   - Run ```set config tikv `raftstore.apply-pool-size`=2;``` to decrease the pool size
   - Run ```show config where type='tikv' and name='raftstore.store-pool-size';``` to check current configuration
   - Run ```ps -T -p `pidof tikv-server` | grep raftstore- to check poller threads-``` No code

Side effects

- No

### Release note 

```release-note
raftstore: support dynamically scaling pool size (#11159)
```
